### PR TITLE
Feature/boas 469 implement auth for internal api

### PR DESCRIPTION
**Pull request by LetsMakeItTechnical**
## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [ ] I have performed a self-review of my own code
- [ ] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [ ] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [ ] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/api/src/server/applications/application/documents/__tests__/delete-document.test.js [hunk 1]
```diff
@@ -1,83 +1,199 @@
...
-		where: {
-			guid: '1111-2222-3333'
-		}
+	test('The test case involves deleting a document setting its "isDeleted" property to "true". when enabling clientCredentialsGrantEnabled', async () => {
+		// GIVEN
+
+		// @ts-ignore
+		jest.replaceProperty(config, 'clientCredentialsGrantEnabled', true);
+
+		jwt.decode.mockResolvedValue({
```
[the remark this answers is not in the document]
Comment (LetsMakeItTechnical) [reply apps/api/src/server/applications/application/documents/__tests__/delete-document.test.js:64]: Ill just test the =function itself

**Review by mayya-bond-kc — DISMISSED** [review]
LGTM. Do we need to add something to the .env.development files to make this work locally?

File: apps/web/.env.example [hunk 2]
```diff
@@ -11,6 +11,10 @@
 # This secret is provided by the Azure AD during app registration.
 # AUTH_CLIENT_SECRET=
 
+# The unique application (api) uuid assigned to the web app by Azure AD when
+# the app was registered.
+# AUTH_API_CLIENT_ID=
```
Comment (francescobrenci-AR) [inline apps/web/.env.example:16]: to work locally the web app needs this value set in the env.local

Comment (francescobrenci-AR) [reply apps/web/.env.example:16]: unless we disable it locally for both WEB and API as we do for Auth

Comment (LetsMakeItTechnical) [reply apps/web/.env.example:16]: @francescobrenci-AR  I can't see this file

**Review by mayya-bond-kc — APPROVED** [review]

**Review by mayya-bond-kc — APPROVED** [review]

**Pull request merged**